platform/security/filter/security-filter-csrf/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- /**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/ -->
+<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0">
+
+  <reference id="securityLogger" interface="ddf.security.audit.SecurityLogger" availability="optional"/>
+
+  <bean id="filter" class="org.codice.ddf.security.filter.csrf.CsrfFilter">
+    <argument ref="securityLogger"/>
+    <property name="whiteListContexts">
+      <array value-type="java.lang.String">
+        <value>/services[/]?$=GET</value>
+        <value>/services/admin/config[/]?$=GET</value>
+        <value>/services/content[/]?$=GET</value>
+        <value>/services/catalog/query[/]?$=GET</value> <!-- Simple Search -->
+        <value>/services/catalog/sources.*=GET</value> <!-- REST Action providers -->
+        <value>/services/internal/catalog/download/cache.*=GET</value>
+        <value>/services/internal/session.*=GET</value>
+        <value>/services/internal/metrics.*=GET</value>
+        <value>/services/logout/actions[/]?$=GET</value>
+        <value>/services/logout/local[/]?$=GET</value>
+        <value>/services/platform/config/ui[/]?$=GET</value>
+        <value>/services/saml/logout[/]?$=GET</value>
+        <value>/services/saml/logout/request[/]?$=GET</value>
+        <value>/services/saml/sso[/]?$=POST</value>
+        <value>/services/saml/sso/metadata[/]?$=GET</value>
+        <value>/services/store/config[/]?$=GET</value>
+        <value>/services/internal/registries.*=GET</value>
+        <value>/services/oidc/logout.*=GET</value>
+      </array>
+    </property>
+  </bean>
+
+  <service ref="filter" interface="org.codice.ddf.platform.filter.SecurityFilter" ranking="101">
```

Comment:
❓ Where does this place the CSRF filter in the order? Before the auth filters?

Reply:
That is my understanding after the websso and login filter but let me double-check